static/app/components/searchQueryBuilder/tokens/filter/aggregateKey.tsx

```diff
@@ -34,10 +34,22 @@ export function AggregateKey({
   filterRef,
 }: AggregateKeyProps) {
   const ref = useRef<HTMLDivElement>(null);
-  const {disabled} = useSearchQueryBuilder();
+  const {dispatch, focusOverride, disabled} = useSearchQueryBuilder();
 
   const [isEditing, setIsEditing] = useState(false);
 
+  useLayoutEffect(() => {
```

Comment:
It might be my lack of familiarity with the search query builder, but what is this hook doing?

Reply:
No you're right it is not obvious at all - added a comment to explain that it enters edit mode when the focus override is set to this item (the focus override is used when we specifically want to set the focus elsewhere after a state change)